src/main/resources/Office365/Groovy.xml

```diff
@@ -223,12 +222,12 @@ public class AzureAuth {
      def qs = request.getQueryString();
      if (qs!=null &amp; qs!="")
       finalRedirect += "?" + qs;
-     state = finalRedirect.hashCode().toString();
-     storedStates.put(state, finalRedirect.replaceAll("clearAuth=1",""));
+     finalRedirect = finalRedirect.replaceAll("clearAuth=1","");
+     request.getSession().setAttribute(AZURE_SESSION_REDIRECT, finalRedirect);
    }
 
-   def currentURL = URLEncoder.encode(REDIRECT_URI + "?state=${state}", "UTF-8")
-   def redirectUrl = "${authority}/${tenant}/oauth2/authorize?response_type=code%20id_token&amp;scope= Files.ReadWrites%20User.Read%20User.ReadWriteUser.ReadBasic.All%20Files.ReadWrite%20Files.ReadWrite.All%20Files.ReadWrite.AppFolder%20Sites.ReadWrite.All&amp;response_mode=form_post&amp;redirect_uri=${currentURL}&amp;state=${state}&amp;client_id=${azureClientID}&amp;nonce=${UUID.randomUUID()}";
+   def currentURL = URLEncoder.encode(REDIRECT_URI, "UTF-8")
+   def redirectUrl = "${authority}/${tenant}/oauth2/authorize?response_type=code%20id_token&amp;scope= Files.ReadWrites%20User.Read%20User.ReadWriteUser.ReadBasic.All%20Files.ReadWrite%20Files.ReadWrite.All%20Files.ReadWrite.AppFolder%20Sites.ReadWrite.All&amp;response_mode=form_post&amp;redirect_uri=${currentURL}&amp;client_id=${azureClientID}&amp;nonce=${UUID.randomUUID()}";
```

Comment:
Please make sure to keep the line length below or equal to 120, as it is hard to read and introduces the need of horizontal scroll. You can split the big expressions, urls into smaller parameters.

Reply:
Fixed in https://github.com/xwikisas/application-office365/pull/11/commits/73ead5a9b328d8ac9d7dd47a720688076251f346 . thanks to check.